drivers/frequency/adf4368/iio_adf4368.c

```diff
@@ -0,0 +1,1199 @@
+	return iio_format_value(buf, len, IIO_VAL_INT_PLUS_MICRO, 2, val);
+}
+
+/**
+ * @brief Handles the write request for the charge pump current attribute.
+ * @param dev     - The iio device structure.
+ * @param buf	  - Command buffer to be filled with the data to be written.
+ * @param len     - Length of the received command buffer in bytes.
+ * @param channel - Command channel info.
+ * @param priv    - Command attribute id.
+ * @return 	  - Result of the writing procedure, error code otherwise.
+ */
+static int adf4368_iio_write_cpi(void *dev, char *buf, uint32_t len,
+				 const struct iio_ch_info *channel,
+				 intptr_t priv)
+{
+	struct adf4368_iio_dev *iio_adf4368 = (struct adf4368_iio_dev *)dev;
+	struct adf4368_dev *adf4368;
+	int32_t val[2];
+	int index;
+
+	if (!iio_adf4368)
+		return -EINVAL;
+
+	adf4368 = iio_adf4368->adf4368_dev;
+
+	if (!adf4368)
+		return -EINVAL;
+
+	iio_parse_value(buf, IIO_VAL_INT_PLUS_MICRO, &val[0], &val[1]);
+
+	index = adf4368_iio_find_2d_row(ci_table, NO_OS_ARRAY_SIZE(ci_table),
+					val[0], val[1]);
+	if (index < 0)
+		return index;
+
+	return adf4368_set_cp_i(adf4368, index);
+}
+
+/**
+ * @brief Handles the read request for the bleed current attribute.
+ * @param dev     - The iio device structure.
+ * @param buf	  - Command buffer to be filled with requested data.
+ * @param len     - Length of the received command buffer in bytes.
+ * @param channel - Command channel info.
+ * @param priv    - Command attribute id.
+ * @return 	  - The size of the read data in case of success, error code
+ * 		    otherwise.
+ */
+static int adf4368_iio_read_bleed_i(void *dev, char *buf, uint32_t len,
+				    const struct iio_ch_info *channel,
+				    intptr_t priv)
+{
+	struct adf4368_iio_dev *iio_adf4368 = (struct adf4368_iio_dev *)dev;
+	struct adf4368_dev *adf4368;
+	int32_t val;
+	int ret;
+
+	if (!iio_adf4368)
+		return -EINVAL;
+
+	adf4368 = iio_adf4368->adf4368_dev;
+
+	if (!adf4368)
+		return -EINVAL;
+
+	ret = adf4368_get_bleed_word(adf4368, &val);
+	if (ret)
+		return ret;
+
+	return iio_format_value(buf, len, IIO_VAL_INT, 1, &val);
+}
+
+/**
+ * @brief Handles the write request for the bleed current attribute.
+ * @param dev     - The iio device structure.
+ * @param buf	  - Command buffer to be filled with the data to be written.
+ * @param len     - Length of the received command buffer in bytes.
+ * @param channel - Command channel info.
+ * @param priv    - Command attribute id.
+ * @return 	  - Result of the writing procedure, error code otherwise.
+ */
+static int adf4368_iio_write_bleed_i(void *dev, char *buf, uint32_t len,
+				     const struct iio_ch_info *channel,
+				     intptr_t priv)
+{
+	struct adf4368_iio_dev *iio_adf4368 = (struct adf4368_iio_dev *)dev;
+	struct adf4368_dev *adf4368;
+	int32_t val;
+
+	if (!iio_adf4368)
+		return -EINVAL;
+
+	adf4368 = iio_adf4368->adf4368_dev;
+
+	if (!adf4368)
+		return -EINVAL;
+
+	iio_parse_value(buf, IIO_VAL_INT, &val, NULL);
```

Comment:
i know this is not going to fail as it is now, but please check the return code of this function everywhere you use it, somebody may add failure points to it in the future and it can silently fail if it's not properly checked

Reply:
Fixed